File: Data Structures & Algorithms/merge-strings-alternately/submission-1.js
class Solution {
    /**
     * @param {string} word1
     * @param {string} word2
     * @return {string}
     */
    mergeAlternately(word1, word2) {
        let res = "";
        if (word1.length <= word2.length) {
            let cur = 0;
            for (let i = 0; i < word1.length; ++i) {
                cur++;
                res += word1[i] + word2[i];
            }

            res += word1.slice(cur);
        } else {
            let cur = 0;
            for (let i = 0; i < word2.length; ++i) {
                cur++;
                res += word1[i] + word2[i];
            }

            res += word2.slice(cur);
        }

        return res;
    }
}

File: Data Structures & Algorithms/plus-one/submission-1.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if digits[len(digits) - 1] < 9:
            digits[len(digits) - 1] += 1
            return digits
        else:
            digits.append(0)
            print(digits)
            cur = len(digits) - 2
            print(cur)
            while digits[cur] == 9 and cur > 0:
                digits[cur] = 0
                cur -= 1

            print(cur)
            if digits[cur] == 9:
                digits[cur] = 1
            else:
                digits[cur] += 1
            digits[cur] = 1
            print(digits)
        return digits

